mods/mc_student/init.lua

```diff
@@ -265,38 +348,43 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 	end
 
 	if formname == "mc_student:marker" then
-		if fields.back then
-			show_student_menu(player)
-		elseif fields.message then
+		if fields.message then
 			place_marker(player,fields.message)
 		elseif fields.message == nil then
 			return true
 		end
 	end
 
 	if formname == "mc_student:coordinates" then
-		if fields.back then
-			show_student_menu(player)
-		elseif fields.record then
-			record_coordinates(player,fields.message)
+		local event = minetest.explode_textlist_event(fields.coordlist)
+
+		if fields.record then
+			record_coordinates(player,fields.note)
 		elseif fields.coordlist then
-			local event = minetest.explode_textlist_event(fields.coordlist)
-		    	if event.type == "CHG" then
-			-- "CHG" = something is selected in the list
-				context.selected = event.index
-		    	end
+			if event.type == "CHG" then
+				selectedCoord = event.index
+			end
 		elseif fields.go then
-			local pname = player:get_player_name()
-			local pmeta = player:get_meta()
-			if not context.selected then
-				context.selected = 1
+			if selectedCoord ~= 0 then
+				local new_pos = minetest.deserialize(pmeta:get_string("coordinates")).coords[selectedCoord]
+				player:set_pos(new_pos)
```

Comment:
There should be a check here first to ensure that the realm still exists (has not been deleted) and that the student is still authorized to join the realm (i.e., either registered in the realm or it is a spawn realm or it is a default realm). Want to avoid allowing students to get into realms they shouldn't be in and also teleporting them to an empty location. See comment above at line 179.

Reply:
Also, using noclip, I was able to move outside the physical bounds of the realm and still record a coordinate and then teleport to that coordinate. Can we use map.conf data to limit the logical coordinates? Just trying to avoid this tool being used nefariously since it allows you to teleport anywhere in the world.